pkg/clients/device/device.go

```diff
@@ -45,7 +45,24 @@ func CreateFromDevice(d *v1alpha1.Device) *packngo.DeviceCreateRequest {
 	}
 }
 
-// NeedsUpdate ... TODO
+// NeedsUpdate returns true if the supplied Kubernetes resource differs from the
+// supplied Packet resource. It considers only fields that can be modified in
+// place without deleting and recreating the instance.
```

Comment:
Hmm I wasn't aware that other fields would cause delete and recreate. I agree that we do not want to do that in normal reconciliation. However, I think we would want to reflect somehow that the external resource has drifted from the desired spec but is not going to be changed by the controller.

Reply:
We are looking at addressing this issue here crossplaneio/crossplane-runtime#39

Reply:
I think that part of that comment I pasted when I used other providers as a template.

Reply:
Oh gotcha! Do you think that we should check every field that can be updated as part of [`DeviceUpdateRequest`](https://github.com/packethost/packngo/blob/6292a62434aeffca0d2cca5c2079349c2fc64650/devices.go#L166) then?